app/models/custom_button.rb

```diff
@@ -157,6 +157,10 @@ def self.button_classes
     BUTTON_CLASSES.collect(&:name)
   end
 
+  def visible_for_current_user
+    visibility.key?(:roles) && (visibility[:roles].include?(User.current_user.miq_user_role_name) || visibility[:roles].include?("_ALL_"))
```

Comment:
maybe explicitly `return false unless visibility.key?(:roles)`

and change the name to `visible_for_current_user?`